bukkit/src/main/java/com/denizenscript/depenizen/bukkit/events/magicspells/ManaChangeScriptEvent.java

```diff
@@ -0,0 +1,127 @@
+package com.denizenscript.depenizen.bukkit.events.magicspells;
+
+import com.nisovin.magicspells.events.ManaChangeEvent;
+import net.aufdemrand.denizen.BukkitScriptEntryData;
+import net.aufdemrand.denizen.events.BukkitScriptEvent;
+import net.aufdemrand.denizen.objects.dPlayer;
+import net.aufdemrand.denizen.utilities.DenizenAPI;
+import net.aufdemrand.denizencore.objects.Element;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizencore.scripts.ScriptEntryData;
+import net.aufdemrand.denizencore.scripts.containers.ScriptContainer;
+import net.aufdemrand.denizencore.utilities.CoreUtilities;
+import net.aufdemrand.denizencore.utilities.debugging.dB;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+// <--[event]
+// @Events
+// magicspells player mana change
+//
+// @Regex ^on magicspells [^\s]+ mana change$
+//
+// @Triggers when a player's mana value changes.
+//
+// @Cancellable false
+//
+// @Context
+// <context.old_mana> returns the old amount of mana.
+// <context.new_mana> returns the new amount of mana.
+// <context.max_mana> returns maximum mana the player can have.
+// <context.reason> returns the reason of the change. Can either be POTION, REGEN, SPELL_COST or OTHER
+//
+// @Determine
+// Element(Number) to set a new mana value.
+//
+// @Plugin DepenizenBukkit, MagicSpells
+//
+// -->
```

Comment:
Please put the meta for commands inside the class rather than before it (should be spaced out at at least one tab's worth)